azure-servicebus/src/main/java/com/microsoft/azure/servicebus/primitives/ConnectionStringBuilder.java

```
@@ -330,13 +344,28 @@ public ConnectionStringBuilder setTransportType(final TransportType transportTyp
 		return this;
 	}
 
+    /**
+	 * @return Returns the authentication method.
+	 */
+	public String getAuthentication() {
+		return this.authentication;
+	}
+
+	/**
+	 * @param authentication The authentication type. Enables Azure Active Directory Managed Identity authentication when set to 'Managed Identity'.
+	 */
+	public void setAadManagedIdentity(String authentication) {
+		this.authentication = authentication;
+	}
+	
 	/**
 	 * Returns an inter-operable connection string that can be used to connect to ServiceBus Namespace
 	 * @return connection string
 	 */
 	@Override
 	public String toString()
 	{
+		this.validate();
```

This is against single responsibility - it's specific to the MSI/RBAC change and the validate method does throw exception which can be painful when debugging.

I understand that this call will help with debugging and fail fast, but the validation may be better placed at the time of setting the value, than at the time of reading the value to print out.